server/sonar-server/src/test/java/org/sonar/server/webhook/ws/WebhookDeliveriesActionTest.java

```diff
@@ -0,0 +1,190 @@
+import org.sonarqube.ws.Webhooks;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.sonar.db.component.ComponentTesting.newProjectDto;
+import static org.sonar.db.webhook.WebhookDbTesting.newWebhookDeliveryDto;
+import static org.sonar.test.JsonAssert.assertJson;
+
+public class WebhookDeliveriesActionTest {
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Rule
+  public UserSessionRule userSession = UserSessionRule.standalone();
+
+  @Rule
+  public DbTester db = DbTester.create(System2.INSTANCE);
+
+  private DbClient dbClient = db.getDbClient();
+  private WsActionTester ws;
+  private ComponentDto project;
+
+  @Before
+  public void setUp() {
+    ComponentFinder componentFinder = new ComponentFinder(dbClient);
+    WebhookDeliveriesAction underTest = new WebhookDeliveriesAction(dbClient, userSession, componentFinder);
+    ws = new WsActionTester(underTest);
+    project = db.components().insertComponent(newProjectDto().setKey("my-project"));
+  }
+
+  @Test
+  public void throw_UnauthorizedException_if_anonymous() {
+    expectedException.expect(UnauthorizedException.class);
+
+    ws.newRequest().execute();
+  }
+
+  @Test
+  public void search_by_component_and_return_no_records() throws Exception {
+    userSession.login().addProjectUuidPermissions(project.uuid(), UserRole.ADMIN);
+
+    Webhooks.DeliveriesWsResponse response = Webhooks.DeliveriesWsResponse.parseFrom(ws.newRequest()
+      .setMediaType(MediaTypes.PROTOBUF)
+      .setParam("componentKey", project.getKey())
+      .execute()
+      .getInputStream());
+
+    assertThat(response.getDeliveriesCount()).isEqualTo(0);
+  }
+
+  @Test
+  public void search_by_task_and_return_no_records() throws Exception {
+    userSession.login().addProjectUuidPermissions(project.uuid(), UserRole.ADMIN);
+
+    Webhooks.DeliveriesWsResponse response = Webhooks.DeliveriesWsResponse.parseFrom(ws.newRequest()
+      .setMediaType(MediaTypes.PROTOBUF)
+      .setParam("ceTaskId", "t1")
+      .execute()
+      .getInputStream());
+
+    assertThat(response.getDeliveriesCount()).isEqualTo(0);
+  }
+
+  @Test
+  public void search_by_component_and_return_records_of_example() throws Exception {
+    WebhookDeliveryDto dto = newWebhookDeliveryDto()
+      .setUuid("d1")
+      .setComponentUuid(project.uuid())
+      .setCeTaskUuid("task-1")
+      .setName("Jenkins")
+      .setUrl("http://jenkins")
+      .setCreatedAt(1_500_000_000_000L)
+      .setSuccess(true)
+      .setDurationMs(10)
+      .setHttpStatus(200);
+    dbClient.webhookDeliveryDao().insert(db.getSession(), dto);
+    db.commit();
+    userSession.login().addProjectUuidPermissions(UserRole.ADMIN, project.uuid());
+
+    String json = ws.newRequest()
+      .setParam("componentKey", project.getKey())
+      .execute()
+      .getInput();
+
+    assertJson(json).isSimilarTo(IOUtils.toString(getClass().getResource("example-deliveries.json")));
```

Comment:
I recenty found out that `ws.getDef().responseExampleAsString()` exists and it is quite convenient